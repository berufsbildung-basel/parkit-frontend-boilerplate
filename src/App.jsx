import { useEffect, useState } from 'react';
import { fetch } from 'cross-fetch';

// TODO make globally available
const baseURL = 'http://localhost:1234'

function App() {
  const [serverMessage, setServerMessage] = useState(null)

  useEffect(() => {
    (async () => {
      console.log('fetching');

      const res = await fetch(`${baseURL}/api/test`)
      console.log(res.status)
      const text = await res.text();
      console.log(text);
      setServerMessage(text);
    })();
  });

  return <>
    <h1>Hello World</h1>
    { serverMessage && <p>{serverMessage}</p> }
  </>
}

export default App
