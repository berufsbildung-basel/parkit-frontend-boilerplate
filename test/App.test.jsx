import { describe, expect, it } from "vitest";
import { render, screen } from "@testing-library/react";
import App from "../src/App.jsx";
import { rest } from "msw";
import TestRenderer from "react-test-renderer";
import { developmentBaseUrl } from "../mock/conf.js";
import SwaggerClient from "swagger-client";
import spec from "@berufsbildung-basel/parkit-spec/api.yml";

const baseUrl = developmentBaseUrl;

describe("Example test", () => {
  it("Fetches content from the mock API", async () => {
    render(<App />);

    await screen.findByText("someuser@adobe.com");
  });

  it("Handles error response", async (ctx) => {
    render(<App />);

    ctx.server.use(
      rest.get(`${baseUrl}/api/users`, (req, res, ctx) => {
        return res.once(ctx.status(403, "Forbidden"));
      })
    );

    await expect(screen.findByText("someuser@adobe.com")).rejects.toThrow();
    await screen.findByText("An error occurred: 403 - Forbidden");
  });

  it("Example Snapshot Test", async () => {
    let app;
    await TestRenderer.act(async () => {
      app = TestRenderer.create(<App />);
    });

    expect(app.toJSON()).toMatchSnapshot();
  });

  it("Example POST request", async () => {
    const client = await new SwaggerClient({
      spec: {
        ...spec,
        servers: [
          { url: `${developmentBaseUrl}/api`, description: "Test server" },
        ],
      },
      requestInterceptor: (request) => {
        request.headers["authorization"] = `Basic ${btoa(
          "test@adobe.com:testPassword"
        )}`;
        return request;
      },
    });

    client.apis.vehicles
      .createVehicle(
        {},
        {
          requestBody: {
            license_plate_number: "BL12354",
            make: "Tesla",
            model: "Model S",
            vehicle_type: "car",
            ev: true,
          },
        }
      )
      .then((response) => {});
  });
});
