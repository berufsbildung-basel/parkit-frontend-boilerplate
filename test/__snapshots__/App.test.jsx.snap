// Vitest Snapshot v1

exports[`Example test > Example Snapshot Test 1`] = `
[
  <h1>
    Hello World
  </h1>,
  <p>
    someuser@adobe.com
  </p>,
]
`;
